src/utils/utils_git.py

```diff
@@ -0,0 +1,35 @@
+import subprocess
+
+
+def get_commit_by_date(directory_path, date):
+    return (
+        subprocess.check_output(
+            f"cd '{directory_path}' && git rev-parse --abbrev-ref HEAD "
+            "| xargs -I {} "
+            f" git rev-list -1 --before {date} "
+            " {}",
```

Comment:
No te parece más entendible cortar de esta manera? Es decir un comando por línea?

```suggestion
            f"cd '{directory_path}'"
            f"&& git rev-parse --abbrev-ref HEAD"
            f"| xargs -I {} git rev-list -1 --before {date} {}",
```

Reply:
Si, aunque ya modifique esa linea ya que como se agrego un chekout a la rama actual en un inicio, ya no es necesario obtenener la rama alctual y hacer un rev-list del commit por fecha en cada momento